internal/dlsym/dlsym.go

```diff
@@ -0,0 +1,37 @@
+package dlsym
+
+// #cgo LDFLAGS: -ldl
+//
+// #include <stdlib.h>
+// #include <dlfcn.h>
+import "C"
+
+import (
+	"errors"
+	"fmt"
+	"unsafe"
+)
+
+// ErrUndefinedSymbol is returned by LookupSymbol when the requested symbol
+// could not be found.
+var ErrUndefinedSymbol = errors.New("symbol not found")
+
+// LookupSymbol resolves the named symbol from the already dynamically loaded
+// libraries. If the symbol is found, a pointer to it is returned, in case of a
+// failure, the message provided by dlerror() is included in the error message.
+func LookupSymbol(symbol string) (unsafe.Pointer, error) {
+	cSymName := C.CString(symbol)
+	defer C.free(unsafe.Pointer(cSymName))
+
+	// clear dlerror before looking up the symbol
+	C.dlerror()
+	// resolve the address of the symbol
+	sym := C.dlsym(nil /* C.RTLD_DEFAULT */, cSymName)
```

Comment:
On MacOS this must be set to `RTLD_DEFAULT`, `NULL` doesn't work.

Reply:
Unfortunately `C.RTLD_DEFAULT` could not be found for some reason. Maybe a header or define is missing, I'll try to figure that out.

Reply:
Added a define for `RTLD_DEFAULT`, it should be in the header, but migth be missing in some Ceph container(s)?

Reply:
That's weird. Would be good to know why. How can dlsym be defined, but RTLD_DEFAULT not? 🤔

Reply:
> That's weird. Would be good to know why. How can dlsym be defined, but RTLD_DEFAULT not? 🤔

Yeah, I was not able to find out. Each `dlfcn.h` that I checked had `RTLD_DEFAULT` without `#ifdef ...` or something around it.

Reply:
#1026 might be the key here.